### PR TITLE
Pv2-2503 Provide multi-academic year learner data

### PR DESCRIPTION
**Pull request by hardik-desai-kcl**

### CODE REVIEW

File: src/SFA.DAS.Payments.MatchedLearner.Application/Repositories/PaymentsDataLockRepository.cs [hunk 1]
```diff
@@ -29,70 +30,72 @@ public async Task<MatchedLearnerDataLockInfo> GetDataLockEvents(long ukprn, long
...
             var stopwatch = Stopwatch.StartNew();
 
-            var latestSuccessfulJob = await _context.LatestSuccessfulJobs
-                .Where(y => y.Ukprn == ukprn)
-                .OrderByDescending(y => y.AcademicYear)
-                .ThenByDescending(y => y.CollectionPeriod)
-                .FirstOrDefaultAsync();
-
-            if (latestSuccessfulJob == null)
+            var latestSuccessfulJobs = _context.LatestSuccessfulJobs
```
Comment (ChughAkshay) [inline src/SFA.DAS.Payments.MatchedLearner.Application/Repositories/PaymentsDataLockRepository.cs:33]: I think the await got cut during refactor

**Review by ChughAkshay — APPROVED** [review]

**Review by AkshayBanifatsyevichChugh — DISMISSED** [review]

File: src/SFA.DAS.Payments.MatchedLearner.Application/Repositories/PaymentsDataLockRepository.cs [hunk 2]
```diff
@@ -29,70 +30,72 @@ public async Task<MatchedLearnerDataLockInfo> GetDataLockEvents(long ukprn, long
...
                 .Where(x =>
                     x.LearningAimReference == "ZPROG001" &&
                     x.Ukprn == ukprn &&
                     x.LearnerUln == uln &&
-                    x.JobId == latestSuccessfulJob.DcJobId &&
-                    x.AcademicYear == latestSuccessfulJob.AcademicYear &&
-                    x.CollectionPeriod == latestSuccessfulJob.CollectionPeriod)
+                    latestSuccessfulJobs.Contains(x.JobId))
                 .OrderBy(x => x.LearningStartDate)
                 .ToListAsync();
```
Comment (abroodie) [inline src/SFA.DAS.Payments.MatchedLearner.Application/Repositories/PaymentsDataLockRepository.cs:58]: Probably worth checking that this query is covered by an index.

File: src/SFA.DAS.Payments.MatchedLearner.Application/Repositories/PaymentsDataLockRepository.cs [hunk 3]
```diff
@@ -29,70 +30,72 @@ public async Task<MatchedLearnerDataLockInfo> GetDataLockEvents(long ukprn, long
...
                 .OrderBy(p => p.StartDate)
+                .ThenBy(p => p.PriceEpisodeIdentifier)
                 .ToListAsync();
 
-
             var dataLockEventPayablePeriods = await _context.DataLockEventPayablePeriod
-                .Where(d => eventId.Contains(d.DataLockEventId) && transactionTypes.Contains(d.TransactionType) && d.PriceEpisodeIdentifier != null && d.Amount != 0)
+                .Where(d => eventIds.Contains(d.DataLockEventId) && transactionTypes.Contains(d.TransactionType) && d.PriceEpisodeIdentifier != null && d.Amount != 0)
                 .OrderBy(p => p.DeliveryPeriod)
                 .ToListAsync();
```
Comment (abroodie) [inline src/SFA.DAS.Payments.MatchedLearner.Application/Repositories/PaymentsDataLockRepository.cs:71]: As this table is quite large probably worth checking this is covered by an index too.

**Review by abroodie — APPROVED** [review]

**Review by tom-gough — DISMISSED** [review]
Hardik approved my latest changes but cannot submit the review himself as he raised the original PR

**Review by tom-gough — APPROVED** [review]

**Review by tom-gough — APPROVED** [review]

**Pull request merged**